xilem_web/src/app.rs

```diff
@@ -1,27 +1,32 @@
 // Copyright 2023 the Xilem Authors
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{DomNode, ViewCtx};
+use crate::{elements::DomChildrenSplice, AnyPod, DomFragment, ViewCtx};
 use std::{cell::RefCell, rc::Rc};
 
-use crate::{DomView, DynMessage, PodMut};
-use xilem_core::{MessageResult, ViewId};
+use crate::DynMessage;
+use wasm_bindgen::UnwrapThrowExt;
+use xilem_core::{AppendVec, MessageResult, ViewId};
 
 pub(crate) struct AppMessage {
     pub id_path: Rc<[ViewId]>,
     pub body: DynMessage,
 }
 
 /// The type responsible for running your app.
-pub struct App<T, V: DomView<T>, F: FnMut(&mut T) -> V>(Rc<RefCell<AppInner<T, V, F>>>);
+pub struct App<State, Fragment: DomFragment<State>, InitFragment>(
```

Comment:
I'm not sure about the name `InitFragment` for the `app_logic`, I'm open for other suggestions, it's also related to [this](https://xi.zulipchat.com/#narrow/stream/354396-xilem/topic/.22app_logic.22)